modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/viewmodel/PodcastsViewModel.kt

```diff
@@ -316,24 +319,19 @@ class PodcastsViewModel
         }
     }
 
-    private fun List<SuggestedFolder>.toFolders(): List<SuggestedFolderModel> {
-        val grouped = this.groupBy { it.name }
-
-        return grouped.map { (folderName, folderItems) ->
-            SuggestedFolderModel(
-                name = folderName,
-                podcasts = folderItems.map { it.podcastUuid },
-                color = 1,
-            )
-        }
-    }
-
     fun showSuggestedFoldersPaywallOnOpen(isSignedInAsPlusOrPatron: Boolean) =
         FeatureFlag.isEnabled(Feature.SUGGESTED_FOLDERS) && !isSignedInAsPlusOrPatron && settings.suggestedFolderPaywallDismissTime.value == 0L
 
     sealed class SuggestedFoldersState {
+        data object Idle : SuggestedFoldersState()
         data object Loading : SuggestedFoldersState()
-        data class Loaded(val folders: List<SuggestedFolderModel>) : SuggestedFoldersState()
+        data class Loaded(private val folders: List<SuggestedFolder>) : SuggestedFoldersState() {
+            private val convertedFolders: List<SuggestedFolderModel> by lazy {
+                folders.toFolders()
+            }
```

Comment:
I don't quite understand, does this need to be lazy as it isn't an expensive operation?

Reply:
We don't have to use lazy, but I wanted to optimize the converting effort. I am going to merge this PR because I need to get all current in review merged. But if you feel I should remove the lazy and can do this in another PR